packages/transaction-controller/src/TransactionController.ts

```diff
@@ -645,19 +704,21 @@ export class TransactionController extends BaseController<
       const chainId = parseInt(currentChainId, undefined);
       const { approved: status } = TransactionStatus;
 
-      const txNonce =
-        nonce ||
-        (await query(this.ethQuery, 'getTransactionCount', [from, 'pending']));
+      nonceLock = await this.nonceTracker.getNonceLock(from);
+
+      // if a nonce already exists on the transactionMeta it means this is a speedup or cancel transaction
+      // so we want to reuse that nonce and hope that it beats the previous attempt to chain. Otherwise use a new locked nonce
+      const nonceToUse = nonce ?? nonceLock.nextNonce;
 
       transactionMeta.status = status;
-      transactionMeta.transaction.nonce = txNonce;
+      transactionMeta.transaction.nonce = addHexPrefix(nonceToUse.toString(16));
```

Comment:
Hmm, this is a little confusing because `nonceToUse` is already a string if we're using the pre-existing `nonce` parameter. Maybe we can update this so we're only stringifying it when we need to.